react-16/src/Form.js

```diff
@@ -0,0 +1,87 @@
+import React, {useState} from 'react';
+
+const Page1 = (props) => {
+
+  const {username, password, setUsername, setPassword, pageStyle} = props;
+
+  return (
+      <div style={pageStyle}>
+        <div>
+          <p>username: </p>
+          <input value={username}
```

Comment:
You don't need a separate update function for each field.
You can add a `name` field to the input and do this:
```js
const handleChange = event => {
  const name = event.target.name
  const value = event.target.value

  setForm(oldForm => ({ ...oldForm, [name]: value }))
}
```